lib/membrane/rtp/session_bin.ex

```diff
@@ -301,6 +312,23 @@ defmodule Membrane.RTP.SessionBin do
       |> to_bin_output(pad)
     ]
 
+    {new_children, new_links} =
+      if extensions == [:vad] do
+        new_children = Map.merge(new_children, %{{:vad, ssrc} => RTP.VAD})
+
+        new_links = [
+          link(:ssrc_router)
+          |> via_out(Pad.ref(:output, ssrc))
+          |> to(rtp_stream_name)
+          |> to({:vad, ssrc})
+          |> to_bin_output(pad)
+        ]
+
+        {new_children, new_links}
+      else
+        {new_children, new_links}
+      end
+
```

Comment:
Tried to write macro
```elixir
defmacro add_extension_link(prev, child, enabled) do
  if enabled do
    quote do
      to(unquote(prev), unquote(child))
    end
  else
    quote do
      unquote(prev)
    end
  end
end
```

It should add link to `child` if `enabled` is true, otherwise it should just return what it got but that's not how macros work

Reply:
The problem is that if enabled do` will be evaluated in the compile-time, so `enabled` won't be a boolean, but AST of whatever is passed as `enabled`. It may work If you just make it a function.